core/services/ocr3/plugins/ccip_integration_tests/home_chain/home_chain_test.go

```diff
@@ -0,0 +1,238 @@
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	logger2 "github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/p2pkey"
+	helpers "github.com/smartcontractkit/chainlink/v2/core/services/ocr3/plugins/ccip_integration_tests"
+	libocrtypes "github.com/smartcontractkit/libocr/ragep2p/types"
+
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	chainA  uint64 = 1
+	fChainA uint8  = 1
+
+	chainB  uint64 = 2
+	fChainB uint8  = 2
+
+	chainC  uint64 = 3
+	fChainC uint8  = 3
+)
+
+func TestHomeChainReader(t *testing.T) {
+	const (
+		ContractName      = "CCIPCapabilityConfiguration"
+		FnGetChainConfigs = "getAllChainConfigs"
+	)
+	// Initialize chainReader
+	cfg := evmtypes.ChainReaderConfig{
+		Contracts: map[string]evmtypes.ChainContractReader{
+			ContractName: {
+				ContractABI: capcfg.CCIPCapabilityConfigurationMetaData.ABI,
+				Configs: map[string]*evmtypes.ChainReaderDefinition{
+					FnGetChainConfigs: {
+						ChainSpecificName: FnGetChainConfigs,
+					},
+				},
+			},
+		},
+	}
+	//============================Setup Backend===================================
+	transactor := testutils.MustNewSimTransactor(t)
+	backend := backends.NewSimulatedBackend(core.GenesisAlloc{
+		transactor.From: {Balance: assets.Ether(1000).ToInt()},
+	}, 30e6)
+	//==============================Setup Contracts - Add capabilities=================================
+	capRegAddress, capRegContract, err := prepareCapabilityRegistry(t, backend, transactor)
+	require.NoError(t, err)
+	capConfAddress, capConfContract, err := prepareCCIPCapabilityConfig(t, backend, transactor, capRegAddress)
+	require.NoError(t, err)
+	p2pIDS := addCapabilities(t, backend, transactor, capRegContract, capConfAddress)
+	//==============================Apply configs to Capability Contract=================================
+	chainAConf := setupConfigInfo(chainA, p2pIDS, fChainA, []byte{})
+	chainBConf := setupConfigInfo(chainB, p2pIDS[1:], fChainB, []byte{})
+	chainCConf := setupConfigInfo(chainC, p2pIDS[2:], fChainC, []byte{})
+	inputConfig := []capcfg.CCIPCapabilityConfigurationChainConfigInfo{
+		chainAConf,
+		chainBConf,
+		chainCConf,
+	}
+	_, err = capConfContract.ApplyChainConfigUpdates(transactor, nil, inputConfig)
+	require.NoError(t, err)
+	backend.Commit()
+	//================================Setup HomeChainReader===============================
+	chainReader := *helpers.SetupChainReader(t, backend, capConfAddress, cfg, ContractName)
+	require.NoError(t, err)
+	homeChain := ccipreader.NewHomeChainReader(chainReader, logger2.NullLogger, 1*time.Millisecond)
+	err = homeChain.Start(context.Background())
+	require.NoError(t, err)
```

Comment:
```suggestion
	require.NoError(t, homeChain.Start(testutils.Context(t)))
```

Can trim it to one line and use `testutils.Context(t)` to get a context tied to the test execution.